packages/kbn-apm-synthtrace-client/src/lib/logs/index.ts

```diff
@@ -123,6 +132,11 @@ class Log extends Serializable<LogDocument> {
     super.timestamp(time);
     return this;
   }
+
+  deleteField(fieldName: keyof LogDocument) {
```

Comment:
You don"t need this function.
I see that for your use case, you need to override these 2 properties.

`'host.name'` - this can be done with `hostName` function already available
`'data_stream.type'` - I was intentionally not allowing modification of this variable as it was for logs, but with your use case, you can another function just like `hostname` called `type` and use that to override the value.

i am against the `deleteField` function because it is very powerful and incorrect usage of this could cause issues. We should not let consumers delete certain fields and this function will actually let them do that.

Reply:
discussed offline, we will keep this function for now